pages/coupon-fraud/index.js

```diff
@@ -0,0 +1,147 @@
+import {useVisitorData} from "../../client/use-visitor-data";
+import {UseCaseWrapper} from "../../client/components/use-case-wrapper";
+import {useState, useEffect, useRef} from "react";
+import FormControl from "@mui/material/FormControl";
+import clsx from "clsx";
+import TextField from "@mui/material/TextField";
+import Button from "@mui/material/Button";
+import Alert from "@mui/material/Alert";
+import makeStyles from "@mui/styles/makeStyles";
+import styles from './coupon-fraud.module.css';
+
+const useStyles = makeStyles((theme) => ({
+    withoutLabel: {
+        marginTop: theme.spacing(3),
+    },
+}));
+
+export default function CouponFraudUseCase() {
+    const visitorDataQuery = useVisitorData({
+        // Don't fetch visitorData on mount
+        enabled: false,
+    });
+
+    const [couponCode, setCouponCode] = useState();
+    const [price, setPrice] = useState(1119);
+
+    const [authMessage, setAuthMessage] = useState();
+    const [severity, setSeverity] = useState();
+    const [isWaitingForReponse, setIsWaitingForReponse] = useState(false);
+    const [httpResponseStatus, setHttpResponseStatus] = useState();
+
+    const messageRef = useRef();
+
+    useEffect(() => {
+        !isWaitingForReponse && messageRef.current?.scrollIntoView({behavior: 'smooth'});
+    }, [isWaitingForReponse]);
+
+    async function handleSubmit(e) {
+        e.preventDefault();
+        setIsWaitingForReponse(true);
+
+        const fpQuery = await visitorDataQuery.refetch();
+        console.log({status: fpQuery.status, error: fpQuery.error, fpQuery});
+        const {requestId, visitorId} = fpQuery.data;
+
+        const claimData = {
+            couponCode,
+            visitorId,
+            requestId,
+        };
+
+        // Serverside handler for this route is located in api/credential-stuffing/authneticate.js file.
+        const response = await fetch('/api/coupon-fraud/claim', {
+            method: 'POST',
+            body: JSON.stringify(claimData),
+            headers: {
+                'Content-Type': 'application/json',
+                Accept: 'application/json',
+            },
+        });
+
+        const responseJson = await response.json();
+        const responseStatus = response.status;
+        if (responseStatus === 200) {
+            setPrice(1000);
+        }
+        setAuthMessage(responseJson.message);
+        setSeverity(responseJson.severity);
+        setHttpResponseStatus(responseStatus);
+        setIsWaitingForReponse(false);
```

Comment:
In most recent use cases, we've been using the `useMutation` hook, which makes the code more readable. You can check it for example here: `client/api/loan-risk/use-request-loan.js`